omnibus/cookbooks/omnibus-supermarket/recipes/cookbook_test.rb

```diff
@@ -26,13 +26,13 @@
 package_path = node['supermarket']['test']['package_path'] || '/tmp/packages'
 version_to_install = node['supermarket']['test']['version_to_install'] || 'nope_set_VERSION_TO_INSTALL'
 
-package 'supermarket' do
-  case node['platform_family']
-  when 'debian'
-    provider Chef::Provider::Package::Dpkg
+case node['platform_family']
+when 'debian'
+  dpkg_package 'supermarket' do
     source "#{package_path}/supermarket_#{version_to_install}-1_amd64.deb"
```

Comment:
Could we determine the package name and use the `package` resource instead of invoking the platform-specific resources?

Reply:
Aside: seems like that could be a good change to the package resource - allowing it to auto resolve extension if source is provided without one.; Or to accept a platform hash for `source`.   

In any case, no need to change it - t's mostly a minor frustration on my part. We have so much package handling across our products that has to work around this, it seems something that the abstraction should provide.

Reply:
Yea. Another reason to not fiddle with this too much is this recipe is only used in the Test Kitchen testing environment, not in any `supermarket-ctl reconfigure` in production.

I'm willing to poke at this to lessen the frustration, though.

Reply:
I have rediscovered why `dpkg_package` and `rpm_package` were used instead of `package`: Ubuntu (and probably Debian) platforms default to the APT package provider and APT doesn't take a source filename to install.

```
           Chef::Exceptions::Package
           -------------------------
           apt package provider cannot handle source attribute. Use dpkg provider instead

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/omnibus-supermarket/recipes/cookbook_test.rb

            35: package 'supermarket' do
            36:   source package_path_and_filename
            37: end
            38:
```